aten/src/ATen/native/IndexingUtils.h

```diff
@@ -0,0 +1,135 @@
+#include <ATen/ExpandUtils.h>
+#include <ATen/native/TensorIterator.h>
+
+namespace at { namespace native {
+
+[[noreturn]]
+static void invalid_mask(const Tensor & self, int64_t idx, const Tensor & mask, int64_t maskIdx) {
+  std::stringstream ss;
+  ss << "The shape of the mask " << mask.sizes() << " at index " << maskIdx;
+  ss << " does not match the shape of the indexed tensor " << self.sizes();
+  ss << " at index " << idx;
+  AT_INDEX_ERROR(ss.str());
+}
+
+
+static std::vector<Tensor> expandTensors(const Tensor & self, TensorList indices) {
+  // Expands ByteTensor (masks) or BoolTensor (masks) into the equivalent indexing by LongTensors
```

Comment:
This comment is slightly confusing: you refer to `masks` but there is no argument named masks here. Did you mean `indices`?

Reply:
Reading the code, I guess what you mean is, if an element of indices is an index tensor, then just return it directly; otherwise convert if from boolean mask to indices mask.